src/wallet/bip39mnemonic.cpp

```diff
@@ -162,40 +162,39 @@ void BIP39Mnemonic::PBKDF2(void *dk, const void *pw, size_t pwLen, const void *s
     // ...
     // Urounds = hmac_hash(pw, Urounds-1)
 
-    uint8_t s[saltLen + sizeof(uint32_t)];
-    uint32_t i, j, U[hashLen/sizeof(uint32_t)], T[hashLen/sizeof(uint32_t)];
+    using SecureUInt32Vector = SecureVector<uint32_t>;
+
+    SecureByteVector s(saltLen + sizeof(uint32_t));
+    uint32_t i, j;
+    SecureUInt32Vector U(hashLen/sizeof(uint32_t)), T(hashLen/sizeof(uint32_t));
 
     assert(dk != NULL || dkLen == 0);
     assert(hashLen > 0 && (hashLen % 4) == 0);
     assert(pw != NULL || pwLen == 0);
     assert(salt != NULL || saltLen == 0);
     assert(rounds > 0);
 
-    memcpy(s, salt, saltLen);
+    memcpy(s.data(), salt, saltLen);
 
     for (i = 0; i < (dkLen + hashLen - 1)/hashLen; i++) {
         j = be32(i + 1);
-        memcpy(s + saltLen, &j, sizeof(j));
+        memcpy(s.data() + saltLen, &j, sizeof(j));
 
         // U1 = hmac_hash(pw, salt || be32(i))
         CHMAC_SHA512 hmacSha512s((const unsigned char*)pw, pwLen);
-        hmacSha512s.Write(s, sizeof(s));
-        hmacSha512s.Finalize((unsigned char*)U);
-        memcpy(T, U, sizeof(U));
+        hmacSha512s.Write(s.data(), sizeof(s));
```

Comment:
@dartdart26 I think the correct vector size should be specified by `s.size()` rather than `sizeof(s)`.